util/test/paratest.chapcs.py

```diff
@@ -0,0 +1,127 @@
+    for line in strip_quotes(squeue_out):
+        (shared, num_nodes) = line.split(',')
+        if shared == 'no':
+            num_exclusive_nodes += int(num_nodes)
+
+    return num_exclusive_nodes
+
+
+def get_num_non_exclusive_nodes():
+    """
+    Get the number of nodes available for testing on the chapel partition (total -
+    exclusive)
+    """
+    sinfo_cmd = ['sinfo',
+                 '--partition=chapel',
+                 '--noheader',
+                 '--responding',
+                 '--format="%D"']
+    sinfo_out = strip_quotes(utils.run_command(sinfo_cmd))
+    num_online_nodes = int(sinfo_out[0])
+    num_non_exclusive_nodes = num_online_nodes - get_num_exclusive_nodes()
+    return num_non_exclusive_nodes
+
+
+def get_num_shared_jobs_running():
+    """
+    Get the number of shared jobs running on the chapel partition
+    """
+    squeue_cmd = ['squeue',
+                  '--partition=chapel',
+                  '--noheader',
+                  '--format="%h"']
+    squeue_out = strip_quotes(utils.run_command(squeue_cmd))
+    return squeue_out.count('yes')
+
+
+def get_good_nodepara():
+    """
+    Get a "good" nodepara value: use up to 3 for comm=none testing, but limit to
+    1 for comm!=none since that's already oversubscribed.
+
+    TODO: Consider increasing nodepara if no other shared jobs are running. Note
+    that we should wait to do this until everybody is using slurm on chapcs."
+    """
+
+    nodepara = 3
+    if chpl_comm.get() != 'none':
+        nodepara = 1
+#    if get_num_shared_jobs_running() == 0:
+#        nodepara += 1
+    return nodepara
+
+
+def run_paratest():
+    """
+    Run paratest inside an salloc using all nodes that are not reserved
+    exclusively on the chapel partition. Throw `--share --nice` and turn off
+    affinity and limit how many executables can run at once so we play nice with
+    other testing going on.
+    """
+    nodepara = get_good_nodepara()
+    num_free_nodes = get_num_non_exclusive_nodes()
+    para_env = ['-env', 'CHPL_TEST_LIMIT_RUNNING_EXECUTABLES=yes', '-env',
+                'QT_AFFINITY=no', '-env', 'QT_SPINCOUNT=300']
+
+    salloc_cmd = ['salloc',
+                  '--nodes={0}'.format(num_free_nodes),
+                  '--immediate',
+                  '--partition=chapel',
+                  '--share',
+                  '--nice']
+
+    paratest_path = os.path.join(os.path.dirname(__file__), 'paratest.server')
+
+    paratest_cmd = salloc_cmd + [paratest_path] + para_env
+    paratest_cmd += ['-nodepara', str(nodepara)] + sys.argv[1:]
+    print('running "{0}"'.format(' '.join(paratest_cmd)))
+
+
+    start_time = timeit.default_timer()
+    for line in utils.run_live_command(paratest_cmd):
+        sys.stdout.write(line)
+    elapsed = int(timeit.default_timer() - start_time)
+    minutes, seconds = divmod(elapsed, 60)
+    print('paratest took {0} minutes and {1} seconds'.format(minutes, seconds))
+
+run_paratest()
```

Comment:
I've generally been trying to include help flags on all scripts in the repository.

It might also be nice to catch the `--help` flag instead of forwarding it to the paratest, e.g.

```python

if '-h' in sys.argv or '--help' in sys.argv:
    print(__doc__)
    return
```